opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -854,7 +853,17 @@ namespace Opm {
         }
     }
 
-    // static member function
+    
+    const MessageContainer& Eclipse3DProperties::getMessageContainer() const {
```

Comment:
Oh my oh my - these containers have multiplied! This implementation seems wrong; when the `MessageContainer`of the full `Eclipse3DProperties` is bound to the `intGridProperties` member messages added by `GridProperties<double>` will be lost space.

I suggest you add the `MessageContainer` member to the `Eclipse3DProperties` instance itself, and then pass a reference to that instance to the `GridProperty<T>` methods - instead of letting them have a private `MessageContainer`.